provide github-actions powered builds

like done in https://github.com/georchestra/mapstore2-georchestra/pull/196 but adapted for the specificities of this repo..

at some point it'll be necessary to start testing the project without having to install the npm whack-a-mole locally :)

right now if i try building on debian stable/buster which only has nodejs v10, it quickly fails with:
```
npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! Unsupported URL Type "npm:": npm:@geosolutions/html-to-draftjs@1.5.1
```

--- a/.github/workflows/cadastrapp.yml
+++ b/.github/workflows/cadastrapp.yml
@@ -36,11 +36,8 @@ jobs:
     - name: "build"
       run: npm run ext:build
 
-    - name: "copy resulting zip"
-      run: mkdir -p scratch && cp dist/Cadastrapp.zip scratch/Cadastrapp-${{ github.sha }}.zip
-
     - name: "publish zip as artifact"
       uses: actions/upload-artifact@v1
       with:
-        name: Cadastrapp.zip
-        path: scratch/Cadastrapp-${{ github.sha }}.zip
+        name: Cadastrapp-${{ github.sha }}
+        path: dist/Cadastrapp.zip

--- a/.github/workflows/cadastrapp.yml
+++ b/.github/workflows/cadastrapp.yml
@@ -1,0 +1,46 @@
+name: "cadastrapp"
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: "checking out"
+      uses: actions/checkout@v2
+    - name: Checkout submodules
+      run: |
+        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+        git submodule sync --recursive
+        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+
+    - name: "setting up npm"
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+
+    - name: "cache node modules"
+      uses: actions/cache@v1
+      env:
+        cache-name: cache-node-modules
+      with:
+        path: ~/.npm # npm cache files are stored in `~/.npm` on Linux/macOS
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+
+    - name: "install dependencies"
+      run: npm install
+
+    - name: "build"
+      run: npm run ext:build
+
+    - name: "copy resulting zip"
+      run: mkdir -p scratch && cp dist/Cadastrapp.zip scratch/Cadastrapp-${{ github.sha }}.zip
+
+    - name: "publish zip as artifact"
+      uses: actions/upload-artifact@v1
+      with:
+        name: Cadastrapp.zip
+        path: scratch/Cadastrapp-${{ github.sha }}.zip

